inductiva/stellarators/scenarios/coils/coils.py

```diff
@@ -59,11 +60,82 @@ def from_circular_curves(cls, num_field_periods, num_coils, coil_currents,
 
             # Set the non-zero coefficients
-            curve_coefficients[1, 0] = math.cos(toroidal_angle) * major_radius
-            curve_coefficients[1, 1] = math.cos(toroidal_angle) * minor_radius
-            curve_coefficients[3, 0] = math.sin(toroidal_angle) * major_radius
-            curve_coefficients[3, 1] = math.sin(toroidal_angle) * minor_radius
-            curve_coefficients[4, 1] = -minor_radius
+            circular_curve_coefficients(curve_coefficients, toroidal_angle,
+                                        major_radius, minor_radius)
+
+            # Create the Coil object
+            coil = Coil(curve_coefficients, coil_currents[i])
+
+            coils.append(coil)
+
+        # Create StellaratorCoils object
+        stellarator_coils = cls(coils, num_field_periods)
+        return stellarator_coils
+
+    @classmethod
+    def from_random_curves(cls,
+                           num_field_periods,
+                           num_coils,
+                           coil_currents,
+                           max_order=12,
+                           major_radius=1.0,
+                           minor_radius=0.5):
+        """Creates random curves.
+
+        Creates a number of 'num_coils' random curves. The random coils are 
+        created by first creating simple equally spaced coils and then randomly 
+        varying the higher order coefficients of the Fourier series describing 
+        the coils. This ensures that the coils are not created on top of each 
+        other and are sufficiently separated but with random and complex curves.
+
+        Args:
+            num_field_periods (int): Number of magnetic field periods.
+            num_coils (int): The number of coils to be randomly created.
+            coil_currents (list): List of coil currents.
+            max_order (int): Maximum order of the coefficients.
+            major_radius (float): distance from the center of the torus 
+            (the central axis) to the outer edge of the plasma region.
+            minor_radius (float): Radius of the simple initial curves.
+
+        Returns:
+            StellaratorCoils: The created StellaratorCoils instance.
+        """
+
+        # To spread the coils equally (in a torus) over an angle that
+        # is a fraction of 2*pi. 2*num_field_periods*num_coils will be
+        # the final number of coils.
+        angle_factor = 2 * num_field_periods * num_coils
+
+        coils = []
+        for i in range(num_coils):
+            toroidal_angle = (i + 0.5) * (2 * np.pi) / (angle_factor)
+            curve_coefficients = np.zeros((6, max_order + 1))
+
+            # Base coefficients to make sure the coils are separated.
+            circular_curve_coefficients(curve_coefficients, toroidal_angle,
+                                        major_radius, minor_radius)
+
+            # Generate random higher coefficients (decreasing the possible
+            # value to make sure the curve is well behaved)
```

Comment:
Minor rewording, "decreasing the range of the random values to ensure the curve is well behaved"